### PR TITLE
cherry-pick: Add release flag for nightly build tag (#2821)

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Cherry-pick of #2821

## Type of change

- Cherry-pick

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: packaging/pre_build_script_windows.sh [hunk 1]
```diff
@@ -6,3 +6,5 @@ python -m pip install tensorrt==10.0.0b6 tensorrt-${CU_VERSION::4}-bindings==10.
 choco install bazelisk -y
 
 cat toolchains/ci_workspaces/WORKSPACE.win.release.tmpl | envsubst > WORKSPACE
+
+echo "RELEASE=1" >> ${GITHUB_ENV}
```
Comment (peri044) [inline packaging/pre_build_script_windows.sh:10]: How does this relate to setting channel=release in the yml files ?

**Review by peri044 — COMMENTED** [review]
LGTM

File: packaging/pre_build_script_windows.sh (hunk 1 above)
Comment (peri044) [inline packaging/pre_build_script_windows.sh:10]: How does this related to channel=release in the yml files ?

Comment (gs-olive) [reply packaging/pre_build_script_windows.sh:10]: Based on the discussion here: https://github.com/pytorch/TensorRT/pull/2806#discussion_r1591617675, this flag seems to get the package naming correct on the Torch upload index. I do not think it directly relates to the `channel` from the `.yml` files.

**Pull request merged**